db/migrate/20190416075600_create_file_attachment_asset.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+class CreateFileAttachmentAsset < ActiveRecord::Migration[5.2]
+  def change
+    create_table :file_attachment_assets do |t|
+      t.references :file_revision,
+                   foreign_key: { to_table: :file_attachment_file_revisions,
+                                  on_delete: :restrict },
+                   index: true,
+                   null: false
+
+      t.references :superseded_by,
+                   foreign_key: { to_table: :file_attachment_assets,
+                                on_delete: :nullify },
+                   null: true
+
+      t.string :variant, null: false
```

Comment:
We may need to work out what we do with variant in this file attachment context.

With images we use this for all the various file sizes, so it can have quite a variety of values. Whereas for attachments the only scenario where we may have two would be a pdf with a thumbnail and we've have assets for each of those.

I guess we'd maybe want variant to be an enum in that case?

Reply:
Added a variant enum to the model 👍